com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/IHotCodeReplaceProvider.java

```diff
@@ -12,8 +12,8 @@
 package com.microsoft.java.debug.core.adapter;
 
 import java.util.List;
-import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
 
 public interface IHotCodeReplaceProvider extends IProvider {
-    CompletableFuture<List<String>> redefineClasses();
+    void redefineClasses(Consumer<List<String>> consumer);
```

Comment:
In the new design, this is more like an event. Please define the interface in event fashion.
```java
IDisposable onClassRedefined(Consumer callback);
```